lib/pharos/ssh/remote_file.rb

```diff
@@ -147,7 +147,7 @@ def each_line
       private
 
       def test?(flag)
-        @client.exec?("sudo test -#{flag} #{escaped_path}")
+        @client.exec!("(sudo true || exit 1) && (sudo test -#{flag} #{escaped_path} && echo true || exit 0)").strip == "true"
```

Comment:
TOCTOU applies, running `sudo` twice won't cancel out random errors.

Something like `sudo sh -c 'test -e ... && echo true || echo false'` would actually help, though... I'm fairly confident that `test`/`stat` itself is unlikely to fail in ways that leads to destructive false-negatives.